CLIENTS.md

```diff
@@ -0,0 +1,266 @@
+        },
+    },
+
+    # Rekor service selection
+    # "Valid" is defined above to mean the service's validity window is
+    # active and the API version is supported by the client.
+    "rekorTlogConfig": {
+        # EXACT specifies that a client must upload entries to exactly
+        # "count" number of valid logs. Clients must throw an error
+        # if less than "count" logs are valid. Clients should select
+        # logs from the highest available API version, even if "count"
+        # logs are not available.
+        # May also be ANY, meaning the client should select exactly
+        # one valid log. The client can decide how to select it, e.g. random
+        # or round-robin if the client tracks state.
+        # May also be ALL, which should be all valid logs.
+        "selector": "EXACT",
+        # Optional, only when EXACT is specified
+        "count": 2
+    }
+
+    # Timestamp authority URLs
+    # Like Rekor, clients should use the TSA config which dictates
+    # how many TSAs should be used to request timestamps from.
+    "tsaUrls": {
+        "url": "https://oauth2.sigstore.dev/auth",
+        "majorApiVersion": 1,
+        "validFor": {
+            "start": "<UTC timestamp>"
+        },
+    },
+
+    # Timestamp authority service selection    
+    "tsaConfig": {
+        "selector": "ANY"
+    }
+}
+```
+
+## Removing Online Verification and Search
+
+Rekor no longer provides an API for online verification and search. This includes
+the APIs for requesting inclusion proofs by index, by leaf hash and by entry,
+and searching for an entry by artifact hash or identity. In the near future,
+we will spin up a separate service to support search.
+
+Clients must be given the inclusion proof and checkpoint for an entry,
+which must be stored in a bundle.
+
+This should have no impact on clients as inclusion proofs were already
+required in bundles. This will only impact monitors, and the read API
+changes are detailed below.
+
+## C2SP Checkpoints
+
+The checkpoints the log provides will conform to the
+[C2SP checkpoint spec](https://github.com/C2SP/C2SP/blob/main/tlog-checkpoint.md).
+Clients must check that their checkpoint verification implementation properly
+handles these checkpoints, which could include:
+
+* Multiple signatures
+* Optional extension lines
+* Updated key names, which will match the shard URL
+* Key IDs calculated per the [signed note spec](https://github.com/C2SP/C2SP/blob/main/signed-note.md#signatures)
+  * ECDSA and Ed25519 key IDs are based on the spec
+  * For RSA, the signature type is `0xFF` and we append `PKIX-RSA-PKCS#1v1.5`,
+    `key ID = SHA-256(key name || 0x0A || 0xFF || PKIX-RSA-PKCS#1v1.5 || public key)[:4]`
+
+## TrustedRoot With Multiple Logs
+
+Clients must verify that verification works with a TrustedRoot
+with multiple logs with overlapping validity windows. Confirm that
+the client will fetch the correct verification key from the TrustedRoot.
```

Comment:
Fetch correct key by matching logid/validity window?

Reply:
Added. I'd say we use log ID for now. We should later align log ID with how we're computing log IDs for checkpoints.